src/bitops.c

```diff
@@ -16,12 +16,23 @@
 /* Count number of bits set in the binary array pointed by 's' and long
  * 'count' bytes. The implementation of this function is required to
  * work with an input string length up to 512 MB or more (server.proto_max_bulk_len) */
-__attribute__((target("popcnt")))
+#if defined(__x86_64__) && ((defined(__GNUC__) && __GNUC__ > 5) || (defined(__clang__)))
+    #if defined(__has_attribute) && __has_attribute(target)
+        #define HAVE_POPCNT
+        #define ATTRIBUTE_TARGET_POPCNT __attribute__((target("popcnt")))
+    #else
+        #define ATTRIBUTE_TARGET_POPCNT
+    #endif
+#else
+    #define ATTRIBUTE_TARGET_POPCNT
+#endif
+
+ATTRIBUTE_TARGET_POPCNT
```

Comment:
this code chunk of detection is a little big, do you consider moving it into `config.h`?

Reply:
sure, i thought it would be just used here. let's move it.

Reply:
done with https://github.com/redis/redis/pull/13609/commits/92065a6cdde5dffdf0075c01f0bad183efa5726d